src/Mutation/PostObjectDelete.php

```diff
@@ -132,6 +136,18 @@ public static function mutate_and_get_payload( WP_Post_Type $post_type_object, s
 				throw new UserError( sprintf( __( 'The %1$s with id %2$s is already in the trash. To remove from the trash, use the forceDelete input', 'wp-graphql' ), $post_type_object->graphql_single_name, $post_id ) );
 			}
 
+			require_once ABSPATH . 'wp-admin/includes/post.php';
+
+			if ( function_exists( 'wp_check_post_lock' ) ) {
+				$user_id = wp_check_post_lock( $post_id );
+				// If post is locked and the override is not specified, do not allow the edit
+				if ( $user_id && true !== $input['ignoreEditLock'] ) {
+					$user = get_userdata( $user_id );
+					/* translators: %s: User's display name. */
+					throw new UserError( sprintf( __( 'You cannot delete this item. %s is currently editing.', 'wp-graphql' ), esc_html( $user->display_name ) ) );
```

Comment:
I wonder if we mention in the response that the mutation can be re-submitted with `ignoreEditLock => true` to override? 🤔